afkak/producer.py

```diff
@@ -381,7 +381,10 @@ def _send_requests(self, parts_results, requests):
         # payload (topic/partition) level.
         payloads = []
         for (topic, partition), reqs in reqsByTopicPart.items():
-            msgSet = create_message_set(reqs, self.codec)
+            if self.client._api_versions != 0 and isinstance(self.client._api_versions, list):
```

Comment:
yeah it was to check that its not `None` basically but now that i'm thinking about it we can just check if `enable_protocol_version_discovery` is `True` or `False` for the client here instead...

Reply:
update... i just removed the second conditional because it should never be `None` when we make it here since the metadata fetch has already run before we actually make it this far to send the produce request... this is also going to be expanded in some form when we add `record/recordBatch` since we will have to decide here most likely which to use based on what we know from the discovered versions so basically this is going to change again